npm/pkg/dataplane/dataplane.go

```diff
@@ -48,23 +51,28 @@ type NPMEndpoint struct {
 	NetPolReference map[string]struct{}
 }
 
-func NewDataPlane(nodeName string, ioShim *common.IOShim, cfg *Config) (*DataPlane, error) {
+func NewDataPlane(nodeName string, ioShim *common.IOShim, cfg *Config, stopChannel <-chan struct{}) (*DataPlane, error) {
 	metrics.InitializeAll()
 	dp := &DataPlane{
+		Config:         cfg,
 		policyMgr:      policies.NewPolicyManager(ioShim, cfg.PolicyManagerCfg),
 		ipsetMgr:       ipsets.NewIPSetManager(cfg.IPSetManagerCfg, ioShim),
 		endpointCache:  make(map[string]*NPMEndpoint),
 		nodeName:       nodeName,
 		ioShim:         ioShim,
 		updatePodCache: make(map[string]*updateNPMPod),
-		Config:         cfg,
+		stopChannel:    stopChannel,
 	}
 
 	err := dp.ResetDataPlane()
 	if err != nil {
 		klog.Errorf("Failed to reset dataplane: %v", err)
 		return nil, err
 	}
+	// necessary for UTs because of ioshim
+	if !dp.disableGoRoutines {
+		dp.policyMgr.Reconcile(dp.stopChannel)
+	}
```

Comment:
so if this is disabled we're not running reconcile at all? 😕  or am I misunderstanding this

Reply:
yes, this is just for unit testing because we test exec calls, and don't want to intermingle exec calls from reconciling with other exec calls